zero_functional.nim

```diff
@@ -286,15 +297,13 @@ proc inlineExists(ext: ExtNimNode): ExtNimNode {.compileTime.} =
 
 proc inlineFind(ext: ExtNimNode): ExtNimNode {.compileTime.} = 
   let adaptedTest = ext.adapt()
-  let listRef = ext.listRef
-  let index = newIdentNode(indexVariableName) 
-  
+  let resultIdent = ext.res
+  let itIdent = ext.prevItNode()
   ext.node = quote:
     if `adaptedTest`:
-      return some(`listRef`[`index`])
-  let f = quote:
-    return none(`listRef`[0].type)
-  ext.finals.add(f)
+      return some(`itIdent`)
+    else:
+      `resultIdent` = none(`itIdent`.type) # TODO: this should be optimized
```

Comment:
this is ineffecient to set it in every loop - I had no better solution, but maybe the compiler will figure it out and the cost is negligible